src/main/java/org/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,51 @@
+package org.example.onlinebookstore.exception;
+
+import java.util.List;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class CustomGlobalExceptionHandler {
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<List<String>> handleValidationExceptions(
+            MethodArgumentNotValidException exception) {
+        List<String> errors = exception.getBindingResult().getAllErrors().stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .toList();
+        return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(SpecificationNotFoundException.class) // not sure about this
```

Comment:
```suggestion
    @ExceptionHandler(SpecificationNotFoundException.class) 
```
remove the comment and return 400 response here